WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java

```diff
@@ -998,9 +999,46 @@ public void onDataLoaded(boolean isEmpty) {
                                             .add(ReaderPostWebViewCachingFragment.newInstance(blogId, postId), tag)
                                             .commit();
                     }
+
+                    // show snackbar when saving post not in saved posts list
+                    if (isBookmarked && getPostListType() == ReaderPostListType.TAG_FOLLOWED
+                        && (mCurrentTag != null && mCurrentTag.isBookmarked())) {
+                        showBookmarkSnackbar();
+                    }
                 }
             };
 
+    private void showBookmarkSnackbar() {
+        if (!isAdded()) {
+            return;
+        }
+
+        Snackbar.make(getView(), R.string.reader_bookmark_snack_title,
+                AccessibilityUtils.getSnackbarDuration(getActivity())).setAction(R.string.reader_bookmark_snack_btn,
+                new View.OnClickListener() {
+                    @Override public void onClick(View view) {
+                        // if the fragment is part of WPMainActivity we just need to update filter and scroll to
+                        // the toolbar. This implementation assumes that Reader list in WPMainActivity will have a
+                        // toolbar.
+                        if (getActivity() instanceof WPMainActivity) {
```

Comment:
I was considering this idea, but we will still had to check if the parent activity is `WPMainActivity` because of the activity transition animations, so I went with least UI disruption.

But looking at the transitions they produce, even through Original one is a bit more smooth, the one handled at `WPMainActivity` might be better because it's more clear what is going on (New Filter, List Reloads). What do you think?

Orignal one:
[![https://gyazo.com/043a97a8c102f1006e0f9bd4d0e8d92e](https://i.gyazo.com/043a97a8c102f1006e0f9bd4d0e8d92e.gif)](https://gyazo.com/043a97a8c102f1006e0f9bd4d0e8d92e)

Handled in `WPMainActivity` with `overridePendingTransition(0, 0)`
[![https://gyazo.com/a27aedc2b62a5d1650346cc5e8cc0894](https://i.gyazo.com/a27aedc2b62a5d1650346cc5e8cc0894.gif)](https://gyazo.com/a27aedc2b62a5d1650346cc5e8cc0894)

Reply:
I think both look ok. But I like the second one a bit more - as you mentioned, it's more clear what's going on. Wdyt @iamthomasbishop?